### PR TITLE
[feat][doc] add migration tutorials for broker load balancing

### PR DESCRIPTION
**Pull request by Anonymitaet**
## What change does this PR introduce?

This PR adds migration tutorials for broker load balancing (topics in the red box below)

image

## Doc development plan - Read before review

For the current doc of Broker Load Balancer, we want to tackle the following challenges:
- Create new documentation to cover all aspects of the feature (especially for the new extensible load balancer).
- Fill in any content gaps that may exist.
- Consolidate existing documentation into the new IA, as it is currently scattered and disorganized.
- Organize the documentation in a cohesive manner, ensuring it flows logically.

Consequently, we’ve created the following content development plans:
- [Broker Load Balancing | Information Architecture](https://xmind.works/share/dzbyKsnS)  
  This document presents a high-level architecture overview, including how we address content gaps and reconcile existing documentation.

- [Broker Load Balancing | Doc Outline](https://docs.google.com/document/d/1zshyn93XJp8Y-uvYLTnFCrcUWaHxeiTjPqdaUrR6zf0/edit#heading=h.6iz0wf853c5u)
  This document delves into the specifics of the content development plan, providing purpose and context for each section.

- [x] `doc` 

## Ackowledgements

@Demogorgon314  @D-2-Ed many thanks for your technical guidance and great support!

### CODE REVIEW

File: sidebars.json [hunk 1]
```diff
@@ -35,7 +35,8 @@
             "concepts-broker-load-balancing-features",
             "concepts-broker-load-balancing-benefits",
             "concepts-broker-load-balancing-concepts",
-            "concepts-broker-load-balancing-types"
+            "concepts-broker-load-balancing-types",
+            "concepts-broker-load-balancing-migration"
```
Comment (Anonymitaet) [inline sidebars.json:39]: For reviewers: I know here is not the perfect place, but there is no better place to locate this tutorial, so I put all types of docs under the same subject "Broker Load Balancing". I'll adjust it once there is a better solution.

image

**Comment by Anonymitaet** [comment]
The preview looks good:

image

**Review by Demogorgon314 — APPROVED** [review]
LGTM. @heesung-sn Can you help review this PR?

File: docs/concepts-broker-load-balancing-migration.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+id: concepts-broker-load-balancing-migration
+title: Migration
+sidebar_label: "Migration"
+---
+
+Pulsar has 3 types of broker load balancers, that is, simple, modular, and extensible. 
+
+You can perform the following migrations.
+
+Migration|When to use
```
Comment (heesung-sn) [inline docs/concepts-broker-load-balancing-migration.md:11]: Change "when to use" to "Available Shedding Strategy."

File: docs/concepts-broker-load-balancing-migration.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    ```bash
+    cd apache-pulsar-@pulsar:version@/bin
+    ```
+
+2. Set `--config` to `loadManagerClassName` and `--value` to `org.apache.pulsar.broker.loadbalance.impl.ModularLoadManagerImpl`.
+
+    ```bash
+    pulsar-admin update-dynamic-config \
+    --config loadManagerClassName \
+    --value org.apache.pulsar.broker.loadbalance.impl.ModularLoadManagerImpl
```
Comment (heesung-sn) [inline docs/concepts-broker-load-balancing-migration.md:74]: @Demogorgon314 does this loadManagerClassName change work dynamically? I don't think this way is recommended due to potential initilaization errors.

File: docs/concepts-broker-load-balancing-migration.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+
+:::note
+
+The pulsar-admin tool is not supported for this migration.
+
+:::
+
+### Change broker.conf file 
+
+1. [Upgrade the Pulsar cluster](./helm-upgrade.md) to 3.0.x.
```
Comment (heesung-sn) [inline docs/concepts-broker-load-balancing-migration.md:91]: to a version greater or equal to `3.0.0.`

File: docs/concepts-broker-load-balancing-migration.md [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+### Change broker.conf file 
+
+1. Access to the broker.conf file.
+
+    ```bash
+    vim apache-pulsar-@pulsar:version@/conf/broker.conf
+    ``````
+
+2. Change the broker load balancer by setting [loadManagerClassName](https://github.com/apache/pulsar/blob/69d7a2bf14555f11a716a9545c5cf391d8179a27/conf/broker.conf#L1309C20-L1309C20) to `ModularLoadManagerImpl` in the broker.conf file.
```
Comment (heesung-sn) [inline docs/concepts-broker-load-balancing-migration.md:53]: Could you remove the code link as the config file lines are updated frequently?

File: docs/concepts-broker-load-balancing-migration.md [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+
+2. Access to the broker.conf file.
+
+    ```bash
+    vim apache-pulsar-@pulsar:version@/conf/broker.conf
+    ```
+
+3. Change the following settings in the broker.conf file:
+
+   - Update [broker load balancer](./concepts-broker-load-balancing-overview.md) by setting [loadManagerClassName](https://github.com/apache/pulsar/blob/69d7a2bf14555f11a716a9545c5cf391d8179a27/conf/broker.conf#L1309C20-L1309C20) to `ExtensibleLoadManagerImpl`.
```
Comment (heesung-sn) [inline docs/concepts-broker-load-balancing-migration.md:101]: Could you remove the code link as the config file lines are updated frequently?

File: docs/concepts-broker-load-balancing-migration.md [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+
+1. Access to the broker.conf file.
+
+    ```bash
+    vim apache-pulsar-@pulsar:version@/conf/broker.conf
+    ```
+
+2. Change the following settings in the broker.conf file:
+   
+   - Update broker load balancer by setting [loadManagerClassName](https://github.com/apache/pulsar/blob/69d7a2bf14555f11a716a9545c5cf391d8179a27/conf/broker.conf#L1309C20-L1309C20) to ModularLoadManagerImpl
```
Comment (heesung-sn) [inline docs/concepts-broker-load-balancing-migration.md:140]: Could you remove the code link as the config file lines are updated frequently?

File: docs/concepts-broker-load-balancing-migration.md [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+    ```
+
+    :::note
+
+    [TransferShedder](./concepts-broker-load-balancing-concepts.md#bundle-unloading-strategies) is **only supported** in the extensible broker load balancer, so you need to change TransferShedder to other [bundle unloading strategies](./concepts-broker-load-balancing-concepts.md#bundle-unloading-strategies).
+
+    :::
+
+3. Restart the Pulsar cluster. The new settings will take effect after the restart.  
+
```
Comment (heesung-sn) [inline docs/concepts-broker-load-balancing-migration.md:157]: @Demogorgon314 

I think we need to explain how Pulsar optimizes the lookup/assignment during this migration. `Topic Lookup and Assignment Redirection during Migration` Please provide the migration logic as you worked on this part.

For example,
During migration from modular to extensible, the lookup and assignment will be redirected to the brokers with the extensible load balancer.

During migration from extensible to modular, the lookup and assignment will be redirected to the brokers with the modular load balancer.

File: docs/concepts-broker-load-balancing-migration.md (hunk 2 above)
Comment (Anonymitaet) [reply docs/concepts-broker-load-balancing-migration.md:11]: "Available Shedding Strategy" is provided below in a table
image

Here want to show "when to use" to give users reminders that they can use a certain type of migration in a certain case

File: docs/concepts-broker-load-balancing-migration.md (hunk 5 above)
Comment (Anonymitaet) [reply docs/concepts-broker-load-balancing-migration.md:53]: This is a permanlink which will not be changed
image

File: docs/concepts-broker-load-balancing-migration.md [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+1. Access the [pulsar-admin tool](https://pulsar.apache.org/docs/next/reference-cli-tools/).
+
+    ```bash
+    cd apache-pulsar-@pulsar:version@/bin
+    ```
+
+2. Set `--config` to `loadManagerClassName` and `--value` to `org.apache.pulsar.broker.loadbalance.impl.ModularLoadManagerImpl`.
+
+    ```bash
+    pulsar-admin update-dynamic-config \
```
Comment (Demogorgon314) [inline docs/concepts-broker-load-balancing-migration.md:72]:
```suggestion
    pulsar-admin brokers update-dynamic-config \
```

File: docs/concepts-broker-load-balancing-migration.md (hunk 3 above)
Comment (Demogorgon314) [reply docs/concepts-broker-load-balancing-migration.md:74]: Yes, I tried out `SimpleLoadManagerImpl` with `ModularLoadManagerImpl`, and it functions properly even after dynamic changes, but when I switched from `ModularLoadManagerImpl` to `ExtensibleLoadManagerImpl`, it did not work.

File: docs/concepts-broker-load-balancing-migration.md (hunk 4 above)
Comment (Anonymitaet) [reply docs/concepts-broker-load-balancing-migration.md:91]: OK, updated

File: docs/concepts-broker-load-balancing-migration.md (hunk 3 above)
Comment (Anonymitaet) [reply docs/concepts-broker-load-balancing-migration.md:74]: OK, then the doc keep unchanged

File: docs/concepts-broker-load-balancing-migration.md (hunk 9 above)
Comment (Anonymitaet) [reply docs/concepts-broker-load-balancing-migration.md:72]: OK,updated

File: docs/concepts-broker-load-balancing-migration.md (hunk 8 above)
Comment (Anonymitaet) [reply docs/concepts-broker-load-balancing-migration.md:157]: OK, updated in https://github.com/apache/pulsar-site/pull/657/files#diff-2abdb541a434c7250793845da41718b322344e45cc83098c3c57addacc4b5a6aR81 and https://github.com/apache/pulsar-site/pull/657/files#diff-2abdb541a434c7250793845da41718b322344e45cc83098c3c57addacc4b5a6aR122

**Comment by Anonymitaet** [comment]
@heesung-sn do you have more comments? If no, I'll merge this PR.

**Pull request merged**